### PR TITLE
fix: rewards v2 audit fixes

### PR DESCRIPTION
**Pull request by 0xrajath**
# Motivation

As part of the Rewards v2 SigmaPrime audit, we need to tackle issue `ELSC2-07: Operators Can Lock In Current Default Split By Front-Running Changes`

# Modifications

* Moved `split <= ONE_HUNDRED_IN_BIPS` logic into common internal function : `_setOperatorSplit`
* Updated `_setOperatorSplit` and `_getOperatorSplit` for edge case of time after first split initialization, but before activation. 
* Updated tests.

# Result

Rewards v2 audit fixes.

### CODE REVIEW

File: src/contracts/core/RewardsCoordinator.sol [hunk 1]
```diff
@@ -395,13 +393,14 @@ contract RewardsCoordinator is
...
+        require(split <= ONE_HUNDRED_IN_BIPS, "RewardsCoordinator._setOperatorSplit: split must be <= 10000 bips");
         require(
             block.timestamp > operatorSplit.activatedAt,
             "RewardsCoordinator._setOperatorSplit: earlier split not activated yet"
         );
         if (operatorSplit.activatedAt == 0) {
-            // If the operator split has not been initialized yet, set the old split to the default split.
-            operatorSplit.oldSplitBips = defaultOperatorSplitBips;
+            // If the operator split has not been initialized yet, set the old split to `type(uint16).max` as a flag.
+            operatorSplit.oldSplitBips = type(uint16).max;
```
Comment (0xrajath) [inline src/contracts/core/RewardsCoordinator.sol:403]: Decided to go with using `type(uint16).max` instead since the Bips variable will never have that value. And also allows us to proceed without using any additional storage.

**Review by nadir-akhtar — APPROVED** [review]

File: src/test/unit/RewardsCoordinatorUnit.t.sol [hunk 2]
```diff
@@ -595,14 +609,28 @@ contract RewardsCoordinatorUnitTests_setOperatorPISplit is RewardsCoordinatorUni
...
+        // Change default split to check if it is returned before activation
+        rewardsCoordinator.setDefaultOperatorSplit(5000);
+        // Check that the split returns the default split before activation for the first time.
+        assertEq(
+            rewardsCoordinator.defaultOperatorSplitBips(),
+            rewardsCoordinator.getOperatorPISplit(operator),
+            "Operator split is not Default split before Activation for first time"
+        );
+
         cheats.warp(activatedAt);
```
Comment (ypatil12) [inline src/test/unit/RewardsCoordinatorUnit.t.sol:634]: Can we also add a regression test for the scenario described in audit report?

Comment (0xrajath) [reply src/test/unit/RewardsCoordinatorUnit.t.sol:634]: Yup it's already there within the `testFuzz_setOperatorAVSSplitFirstTime` and  `testFuzz_setOperatorPISplitFirstTime` tests.

Comment (0xrajath) [reply src/test/unit/RewardsCoordinatorUnit.t.sol:634]: https://github.com/Layr-Labs/eigenlayer-contracts/pull/987/files#diff-1ddf2647b6d1ee196f902307ebcac00fb15be8dd0c02acd2c89ad686cc94265eR476-R485

**Review by ypatil12 — APPROVED** [review]
